src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultTagHelperCompletionService.cs

```diff
@@ -88,23 +88,37 @@ public override AttributeCompletionResult GetAttributeCompletions(AttributeCompl
                     foreach (var attributeDescriptor in descriptor.BoundAttributes)
                     {
                         UpdateCompletions(attributeDescriptor.Name, attributeDescriptor);
+
+                        if (!string.IsNullOrEmpty(attributeDescriptor.IndexerNamePrefix))
+                        {
+                            UpdateCompletions(attributeDescriptor.IndexerNamePrefix + "...", attributeDescriptor);
+                        }
                     }
                 }
                 else
                 {
-                    var htmlNameToBoundAttribute = descriptor.BoundAttributes.ToDictionary(attribute => attribute.Name, StringComparer.OrdinalIgnoreCase);
+                    var htmlNameToBoundAttribute = new Dictionary<string, BoundAttributeDescriptor>(StringComparer.OrdinalIgnoreCase);
+                    foreach (var attributeDescriptor in descriptor.BoundAttributes)
+                    {
+                        htmlNameToBoundAttribute[attributeDescriptor.Name] = attributeDescriptor;
+
+                        if (!string.IsNullOrEmpty(attributeDescriptor.IndexerNamePrefix))
+                        {
+                            htmlNameToBoundAttribute[attributeDescriptor.IndexerNamePrefix] = attributeDescriptor;
+                        }
+                    }
 
                     foreach (var rule in descriptor.TagMatchingRules)
                     {
                         foreach (var requiredAttribute in rule.Attributes)
                         {
                             if (htmlNameToBoundAttribute.TryGetValue(requiredAttribute.Name, out var attributeDescriptor))
                             {
-                                UpdateCompletions(requiredAttribute.Name, attributeDescriptor);
+                                UpdateCompletions(requiredAttribute.DisplayName, attributeDescriptor);
```

Comment:
Required attributes know how to display their name properly based off of how the required attribute was written. i.e. `attributename-*` results in the required attributes displayname displaying as `attributename-...` to indicate a prefix match. 

For the `BoundAttributeDescriptor` case above we don't have that luxury and must append `...` for dictionary property cases (they always apply prefixed based logic).